python/cudf/cudf/utils/cudautils.py

```diff
@@ -244,14 +254,7 @@ def compile_udf(udf, type_signature):
 
     # Check if we've already compiled a similar (but possibly distinct)
     # function before
-    codebytes = udf.__code__.co_code
-    if udf.__closure__ is not None:
-        cvars = tuple([x.cell_contents for x in udf.__closure__])
-        cvarbytes = dumps(cvars)
-    else:
-        cvarbytes = b""
-
-    key = (type_signature, codebytes, cvarbytes)
+    key = make_cache_key(udf, type_signature)
```

Comment:
Let's also move the 2 comments above into the `make_cache_key` function instead, as a "docstring".